moveit_core/planning_interface/include/moveit/planning_interface/planning_response.h

```diff
@@ -53,7 +53,15 @@ struct MotionPlanResponse
 
   robot_trajectory::RobotTrajectoryPtr trajectory_;
   double planning_time_;
-  moveit_msgs::MoveItErrorCodes error_code_;
+  moveit::core::MoveItErrorCode error_code_;
+  moveit_msgs::RobotState start_state_;
```

Comment:
It is possible to pass an optional `solution_selection_callback` and/or a `stopping_criterion_function` both use the produced MotionPlanResponses to make a decision and these to additional fields might be useful to make that decision. For example 

```cpp
bool stoppingCriterionCallBack(
    moveit_cpp::PlanningComponent::PlanSolutions const& plan_solutions,
    moveit_cpp::PlanningComponent::MultiPipelinePlanRequestParameters const& plan_request_parameters)
  for (auto const& solution : solutions)
  {
      // Stop parallel planning if PILZ found a solution
      if (solution.planner_id_ == "LIN")
      {
        return true;
      }
  }
 return false;
}
```

Reply:
I agree for the `planner_id_`, but all plans share the very same start state, don't they?
Please add a comment, e.g. "planner this solution originates from".

Reply:
That's true, it does not matter for the solution selection. I added it since the MotionPlanResponse message has a similar field and it can be useful when the MotionPlanResponse is processed further independent from moveit_cpp/ the planning_component.  Do you think we should remove it or is a comment sufficient?

Reply:
I'm not sure yet. If the start state is identical for all solutions, saving it here would be a waste of memory.
However, due to different pipeline configs, modifying the start state (or not), the actual start state might differ between solutions (but I don't know by heart). That would be an argument to keep it here...

Reply:
> I'd remove the underscore from the field names but you're right, the API will be broken by this. I can open a PR for moveit_tutorials to fix this. Would MTC be affected?

I'm not only worried about moveit_tutorials or MTC. The problem is that _any_ user code out there accessing those members will be affected by this change! At least we should provide a deprecation period for transition. 
My suggestion is to:
- remove the underscores
- provide additional, deprecated members with the underscore as const references to the new members:
  ```cpp
  moveit_msgs::RobotState start_state;
  [[deprecated("Use start_state instead.")]] const moveit_msgs::RobotState& start_state_;
  ```
  Obviously, those references need to be initialized in the constructor. This will render the struct read-only. So you will need to provide a custom copy constructor and assignment operator as well.